packages/common/src/components/ThemeSwitch/ThemeSwitch.tsx

```diff
@@ -36,11 +36,18 @@ export const ThemeSwitch = ({
   theme: defaultTheme,
   ...props
 }: ThemeSwitchProps) => {
-  const [theme, setTheme] = useState(defaultTheme ?? getTheme);
+  const [theme, setTheme] = useState(defaultTheme ?? null);
```

Comment:
Is there any reason why we doing call getTheme at this point? My understanding is we need to get the theme at this point if we already can. Yes, this could return undefined in the context of a server but if we are on the client, then we can easily get the theme here and avoid a rerender from `setTheme` call within the useLayoutEffect. Let me know what you think